modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCachePartitionExchangeManager.java

```diff
@@ -693,26 +693,56 @@ else if (customMsg instanceof WalStateAbstractMessage
         }
     }
 
-    /**
-     * @param msg Event message.
-     * @param evtType Event type.
-     */
-    private void recordEvent(String msg, int evtType) {
-        GridKernalContext ctx = cctx.kernalContext();
+    /** */
+    private void exchangeFutDone(IgniteInternalFuture<AffinityTopologyVersion> fut, ExchangeActions exchActions) {
```

Comment:
I'd rename this method to something more related to cluster state handling.
There are lots of another logic that is listened to exchange future completion.